lib/Dialect/RTL/Ops.cpp

```diff
@@ -255,6 +288,108 @@ static LogicalResult verifyRTLInstanceOp(RTLInstanceOp op) {
   return success();
 }
 
+StringAttr RTLInstanceOp::getResultName(size_t idx) {
+  SmallString<16> attrName = llvm::formatv("rtl.rname{0}", idx);
+  return getAttrOfType<StringAttr>(attrName.str());
+}
+
+ParseResult parseResultNames(OpAsmParser &p, NamedAttrList &attrDict) {
+  MLIRContext *ctxt = p.getBuilder().getContext();
+  if (p.parseOptionalAttrDict(attrDict))
+    return failure();
+
+  for (size_t i = 0, e = p.getNumResults(); i < e; ++i) {
+    SmallString<16> id = llvm::formatv("rtl.rname{0}", i);
+    Identifier resultNameKey = Identifier::get(id, ctxt);
+    if (attrDict.getNamed(resultNameKey))
+      continue;
+
+    StringAttr resultName = StringAttr::get(p.getResultName(i).first, ctxt);
+    attrDict.push_back(NamedAttribute(resultNameKey, resultName));
+  }
+  return success();
+}
+
+void printResultNames(OpAsmPrinter &p, RTLInstanceOp *op) {
+  SmallVector<StringRef, 8> elideFields = {"instanceName", "moduleName"};
+
+  NamedAttrList attrs = op->getAttrs();
+  for (auto it = attrs.begin(), e = attrs.end(); it < e; ++it) {
+    StringRef attrName = it->first.strref();
+    if (attrName.startswith("rtl.rname")) {
+      auto numberStr = attrName.substr(9);
+      size_t resultIdx;
+      if (numberStr.getAsInteger<size_t>(10, resultIdx))
+        continue;
+
+      SmallString<32> resultNameStr;
+      llvm::raw_svector_ostream tmpStream(resultNameStr);
+      p.printOperand(op->getResult(resultIdx), tmpStream);
+      auto expectedName = it->second.dyn_cast<StringAttr>();
+      if (expectedName &&
+          tmpStream.str().drop_front() == expectedName.getValue()) {
+        elideFields.push_back(attrName);
+      }
+    }
+  }
+  p.printOptionalAttrDict(op->getAttrs(), elideFields);
+}
+
+void RTLInstanceOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
+  NamedAttrList attrs = getAttrs();
+  for (auto it = attrs.begin(), e = attrs.end(); it < e; ++it) {
+    if (it->first.strref().startswith("rtl.rname")) {
+      auto numberStr = it->first.strref().substr(9);
+      size_t resultIdx;
+      if (!numberStr.getAsInteger<size_t>(10, resultIdx))
+        setNameFn(getResult(resultIdx),
+                  it->second.dyn_cast<StringAttr>().getValue());
```

Comment:
This will crash with a null pointer dereference if the `dyn_cast` fails.   Plz ignore it if not a string attribute, or have the verifier enforce that it must be a string attr if present.